packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -242,16 +294,18 @@ export class ConnectionStateHandler {
             // don't want to reset the timer as we still want to wait on original client which started this timer.
             if (client !== undefined
                 && this.handler.shouldClientJoinWrite()
-                && this.prevClientLeftTimer.hasTimer === false
+                && this.waitingForLeaveOp === false
             ) {
                 this.prevClientLeftTimer.restart();
             } else {
                 // Adding this event temporarily so that we can get help debugging if something goes wrong.
                 this.logger.sendTelemetryEvent({
                     eventName: "noWaitOnDisconnected",
-                    inQuorum: client !== undefined,
-                    hasTimer: this.prevClientLeftTimer.hasTimer,
-                    shouldClientJoinWrite: this.handler.shouldClientJoinWrite(),
+                    details: JSON.stringify({
```

Comment:
Is it easier to query/analyze like this or the goal is to reduce the number of fields in kusto?

Reply:
Pro's: Fewer columns, you can easily find the info you want, even in "session" queries that include many different events.

Con's: You have to do add a special incantation to your Kusto query to do any analysis (like aggregation) on specific fields.  It's not hard, just an extra step.

My vision is for a world where we have only a couple dozen columns across our telemetry, and only fields that are broadly applicable across multiple events get their own column.